src/views/home/components/token-list/token-list.view.tsx

```diff
@@ -1,287 +1,269 @@
+    token.symbol.toLowerCase().includes(term.toLowerCase());
+
+  const updateTokenList = (tokensWithBalance: Token[], searchTerm: string) => {
+    const newFilteredTokens = tokensWithBalance.filter(getTokenFilterByTerm(searchTerm));
+
+    setFilteredTokens(newFilteredTokens);
+    setCustomToken({ status: "pending" });
+
+    if (ethersUtils.isAddress(searchTerm) && newFilteredTokens.length === 0) {
+      setCustomToken({ status: "loading" });
+
+      const setBalanceAndListCustomToken = (token: Token) => {
+        getTokenBalance(token, chains.from)
+          .then((balance) => {
+            callIfMounted(() => {
+              setCustomToken((currentCustomToken) =>
+                currentCustomToken.status === "pending"
+                  ? currentCustomToken
+                  : {
+                      status: "successful",
+                      data: { ...token, balance: { status: "successful", data: balance } },
+                    }
+              );
+            });
+          })
+          .catch(() => {
+            callIfMounted(() => {
+              setCustomToken((currentCustomToken) =>
+                currentCustomToken.status === "pending"
+                  ? currentCustomToken
+                  : {
+                      status: "successful",
+                      data: {
+                        ...token,
+                        balance: { status: "failed", error: "Couldn't retrieve token balance" },
+                      },
+                    }
+              );
+            });
+          });
+      };
+
+      void getTokenFromAddress({
+        address: searchTerm,
+        chain: chains.from,
+      })
+        .then(setBalanceAndListCustomToken)
+        .catch(() =>
+          getTokenFromAddress({
+            address: searchTerm,
+            chain: chains.to,
+          })
+            .then(setBalanceAndListCustomToken)
+            .catch(() =>
+              callIfMounted(() => {
+                setCustomToken({
+                  status: "failed",
+                  error:
+                    "The token contract cannot be called on any network. Please make sure the contract is deployed on a network.",
+                });
+              })
+            )
+        );
+    }
+  };
 
-  const onOutsideClick = (event: MouseEvent) => {
-    if (event.target === event.currentTarget) {
-      onClose();
+  const handleSearchInputchange = (value: string): void => {
+    if (tokens) {
+      setSearchInputValue(value);
+      updateTokenList(tokens, value);
+      if (value === "") {
+        setCustomToken({ status: "pending" });
+      }
     }
   };
 
-  const isLoading = customToken.status === "loading";
+  useEffect(() => {
+    if (customToken.status === "successful") {
+      setFilteredTokens([customToken.data]);
+    }
+  }, [customToken]);
 
   useEffect(() => {
     if (inputRef.current) {
       inputRef.current.focus();
     }
   }, []);
```

Comment:
The `inputRef` and this effect can now be removed.

I see though that the autofocus does not work when the modal is initially loaded. It only works when we come back from the token info table view.

Reply:
Yep... I spent some time trying to fix this but I still don't know why it's not working... Will try to give it another try again tomorrow.